evalml/pipelines/components/transformers/imputers/time_series_imputer.py

```diff
@@ -0,0 +1,199 @@
+    }
+    """{
+        "categorical_impute_strategy": ["backwards_fill", "forwards_fill"],
+        "numeric_impute_strategy": ["backwards_fill", "forwards_fill", "interpolate"],
+        "target_impute_strategy": ["backwards_fill", "forwards_fill", "interpolate"],
+    }"""
+    _valid_categorical_impute_strategies = set(["backwards_fill", "forwards_fill"])
+    _valid_numeric_impute_strategies = set(
+        ["backwards_fill", "forwards_fill", "interpolate"]
+    )
+    _valid_target_impute_strategies = set(
+        ["backwards_fill", "forwards_fill", "interpolate"]
+    )
+
+    def __init__(
+        self,
+        categorical_impute_strategy="forwards_fill",
+        numeric_impute_strategy="interpolate",
+        target_impute_strategy="forwards_fill",
+        random_seed=0,
+        **kwargs,
+    ):
+        if categorical_impute_strategy not in self._valid_categorical_impute_strategies:
+            raise ValueError(
+                f"{categorical_impute_strategy} is an invalid parameter. Valid categorical impute strategies are {', '.join(self._valid_numeric_impute_strategies)}"
+            )
+        elif numeric_impute_strategy not in self._valid_numeric_impute_strategies:
+            raise ValueError(
+                f"{numeric_impute_strategy} is an invalid parameter. Valid numeric impute strategies are {', '.join(self._valid_numeric_impute_strategies)}"
+            )
+        elif target_impute_strategy not in self._valid_target_impute_strategies:
+            raise ValueError(
+                f"{target_impute_strategy} is an invalid parameter. Valid target column impute strategies are {', '.join(self._valid_target_impute_strategies)}"
+            )
+
+        parameters = {
+            "categorical_impute_strategy": categorical_impute_strategy,
+            "numeric_impute_strategy": numeric_impute_strategy,
+            "target_impute_strategy": target_impute_strategy,
+        }
+        parameters.update(kwargs)
+        self._all_null_cols = None
+        self._forwards_cols = None
+        self._backwards_cols = None
+        self._interpolate_cols = None
+        self._impute_target = None
+        super().__init__(
+            parameters=parameters, component_obj=None, random_seed=random_seed
+        )
+
+    def fit(self, X, y=None):
+        """Fits imputer to data.
+
+        'None' values are converted to np.nan before imputation and are treated as the same.
+        If a value is missing at the beginning or end of a column, that value will be imputed using
+        backwards fill or forwards fill as necessary, respectively.
+
+        Args:
+            X (pd.DataFrame, np.ndarray): The input training data of shape [n_samples, n_features]
+            y (pd.Series, optional): The target training data of length [n_samples]
+
+        Returns:
+            self
+        """
+        forwards_fill_cats = (
+            ["category", "boolean"]
```

Comment:
I think this has to be anything that's not numeric. Theoretically, someone could pass in a dirty dataset with email features to the regularizer since this will happen early in the process, probably prior to modeling. I think those cases are unlikely but it should be easy to extend what you have now to handle them.